tests-e2e/cypress/support/ui_commands.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+function getLastPostId() : Cypress.Chainable<string> {
+    // # Wait until permanent post.
+    cy.get('#postListContent').should('be.visible');
+    cy.waitUntil(() =>
+        cy.findAllByTestId('postView').
+            last().
+            then((el) => !(el[0].id.includes(':')))
+    );
+
+    // # Get the last post and return its ID.
+    return cy.findAllByTestId('postView').last().should('have.attr', 'id').and('not.include', ':').
+        invoke('replace', 'post_', '');
+}
+Cypress.Commands.add('getLastPostId', getLastPostId);
+
+function postInCurrentChannel(message: string): void {
+    // # Type the message in the textbox and type Enter.
+    cy.findByTestId('post_textbox').clear().type(message).type('{enter}');
+
+    // * Verify that the textbox is empty after sending the message.
+    cy.findByTestId('post_textbox').should('have.text', '');
+}
+Cypress.Commands.add('postInCurrentChannel', postInCurrentChannel);
+
+function exportSlashCommand() : void {
+    // # Post /export (with Esc to make sure that the autocompletion is gone)
+    // in the channel.
+    cy.postInCurrentChannel('/export {esc}');
+}
+Cypress.Commands.add('exportSlashCommand', exportSlashCommand);
+
+function archiveCurrentChannel() : void {
+    // # Click on the header dropdown arrow.
+    cy.get('#channelHeaderDropdownIcon').click();
+
+    // # Select the Archive Channel item in the menu.
+    cy.get('#channelArchiveChannel').click();
+
+    // # Confirm the archival of the channel clicking on the Archive button.
+    cy.get('#deleteChannelModalDeleteButton').click();
+}
+Cypress.Commands.add('archiveCurrentChannel', archiveCurrentChannel);
+
+function unarchiveCurrentChannel() : void {
```

Comment:
The feature to view the unarchived channels (ExperimentalViewArchivedChannels) will need to be enabled for the tests using this function to pass.
Maybe adding a helper function in the `server_api_commands` to change the server config would help (e.g. https://github.com/mattermost/mattermost-webapp/blob/d964c4e850afb6bd2f70c3fa6c08468f696a9b6e/e2e/cypress/support/api/system.js#L58)? Otherwise, since test case 16 in permissions_specs seems to be the only test where viewing an unarchived channel is needed, the setting could be changed within the `before` block of the spec file?

Reply:
So many things i did not have into account! Thank you :) Done in https://github.com/mattermost/mattermost-plugin-channel-export/pull/14/commits/52166724a9f1b4a3da9e26495637cb0a5b442e12.